src/Services/Import/Traits/HandlesEcMediaImport.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Wm\WmPackage\Services\Import\Traits;
+
+use Illuminate\Database\Eloquent\Model;
+
+trait HandlesEcMediaImport
+{
+    /**
+     * Process the entire EC Media import
+     *
+     * @param  array  $data  The original data from Geohub
+     * @param  string  $geometry  The geometry of the related model
+     */
+    public function processEcMediaImport(array $data, string $geometry): void
+    {
+        $transformedData = $this->transformEcMediaData($data);
+        $transformedData['custom_properties']['geometry'] = $geometry;
+
+        $relatedModel = $transformedData['model_type']::find($transformedData['model_id']);
+
+        if (! $relatedModel) {
+            throw new \Exception("Related model not found: {$transformedData['model_type']} with ID {$transformedData['model_id']}");
+        }
+
+        // Get the URL and prepare it
+        $url = $transformedData['url'];
+        if (! filter_var($url, FILTER_VALIDATE_URL)) {
+            $url = 'https://geohub.webmapp.it/storage/'.ltrim($url, '/');
+
+            // validate if the url returns an image content type
+            $contentType = get_headers($url, 1)[0];
+            if (strpos($contentType, 'image') === false) {
+                throw new \Exception("The URL {$url} does not return an image content type. Skipping media import.");
+            }
+        }
+
+        // Check if media with the same geohub_id already exists in the collection
+        $existingMedia = $relatedModel->getMedia('feature_image')
+            ->first(function ($media) use ($transformedData) {
+                return isset($media->custom_properties['geohub_id']) &&
+                    $media->custom_properties['geohub_id'] === $transformedData['custom_properties']['geohub_id'];
+            });
+
+        // If media exists, delete it before adding the new one
+        if ($existingMedia) {
+            $existingMedia->delete();
+        }
+
+        $nameJson = json_decode($data['name'], true);
+        $fileName = is_array($nameJson) ? ($nameJson['it'] ?? reset($nameJson)) : $data['name'];
+
+        $fileName = preg_replace('/[^a-zA-Z0-9_-]/', '_', $fileName);
+        $extension = pathinfo(parse_url($url, PHP_URL_PATH), PATHINFO_EXTENSION);
+        $fileName = $fileName.'.'.($extension ?: 'jpg');
+
+        $mediaItem = $relatedModel->addMediaFromUrl($url)
+            ->usingName($fileName)
+            ->usingFileName($fileName)
+            ->toMediaCollection('default', config('wm-media-library.disk_name'))
+            ->withCustomProperties($transformedData['custom_properties']);
```

Comment:
qui metterei il metodo withCustomProperties prima di toMediaCollection

Reply:
https://spatie.be/docs/laravel-medialibrary/v11/advanced-usage/using-custom-properties